lib/network/default.nix

```diff
@@ -0,0 +1,107 @@
+{ lib }:
+{
+  ipv6 = {
+    /**
+      Normalize an IPv6 address by removing the "::" compressions and padding
+      them with the necessary number of zeros to create the correct address. Converts
+      an address from the string to the list of strings which then can be parsed using
+      `_parseNormalized`. Throws an error when the address is malformed.
+
+      # Type: String -> [ String ]
+    */
+    _normalizeIpv6 =
+      addr:
+      if builtins.match "^[0-9A-Fa-f:]+$" addr == null then
+        throw "${addr} contains malformed characters for IPv6 address"
+      else
+        let
+          doubleColon = "::";
+          piecesMax = 8;
+          pieces = lib.strings.splitString ":" addr;
+          piecesNoEmpty = lib.lists.remove "" pieces;
+          piecesNoEmptyLen = builtins.length piecesNoEmpty;
+          emptyCnt = builtins.length pieces - builtins.length piecesNoEmpty;
```

Comment:
emptyCnt is only used once in line 38, so I think we should move it directly into the local scope. I also think that we should call this emptyCount since everything else is written out and it's only internal anyways.